### PR TITLE
Webhook changes

### PR DESCRIPTION
**Pull request by L0neGamer**
Should work on #122.

### CODE REVIEW

File: src/Discord/Internal/Rest/Interactions.hs [hunk 1]
```diff
@@ -13,7 +13,6 @@ import Discord.Internal.Types.Interactions
 import Network.HTTP.Client.MultipartFormData (PartM, partBS)
 import Network.HTTP.Req ((/:))
```
Comment (yutotakano) [inline src/Discord/Internal/Rest/Interactions.hs:14]: Maybe it's easier to add `(/~)` in the import list here, and replace the uses of `R./~`? The qualified operator isn't the prettiest syntax.

File: src/Discord/Internal/Rest/Webhook.hs [hunk 2]
```diff
@@ -46,16 +45,19 @@ data WebhookRequest a where
   CreateWebhook :: ChannelId -> CreateWebhookOpts -> WebhookRequest Webhook
   GetChannelWebhooks :: ChannelId -> WebhookRequest [Webhook]
   GetGuildWebhooks :: GuildId -> WebhookRequest [Webhook]
-  GetWebhook :: WebhookId -> WebhookRequest Webhook
-  GetWebhookWithToken :: WebhookId -> T.Text -> WebhookRequest Webhook
-  ModifyWebhook :: WebhookId -> ModifyWebhookOpts
+  -- GetWebhook :: WebhookId -> WebhookRequest Webhook
+  GetWebhook :: WebhookId -> Maybe WebhookToken -> WebhookRequest Webhook
+  ModifyWebhook :: WebhookId -> Maybe WebhookToken -> ModifyWebhookOpts
```
Comment (aquarial) [inline src/Discord/Internal/Rest/Webhook.hs:50]: Nice to add these together, `Maybe` fits well.

From <https://discord.com/developers/docs/resources/webhook#webhook-resource> the advantage of the token is that it doesn't require logging into a bot? But the library requires logging in to send anything at all.

File: src/Discord/Internal/Rest/Guild.hs [hunk 3]
```diff
@@ -20,7 +20,7 @@ module Discord.Internal.Rest.Guild
 
 
 import Data.Aeson
-import Network.HTTP.Req ((/:))
+import Network.HTTP.Req ((/:), (/~))
```
Comment (aquarial) [inline src/Discord/Internal/Rest/Guild.hs:23]: Nice library function find!

**Review by aquarial — APPROVED** [review]
Looks good, Some minor merge conflicts with https://github.com/aquarial/discord-haskell/pull/119 and ready to merge

File: src/Discord/Internal/Rest/Webhook.hs (hunk 2 above)
Comment (L0neGamer) [reply src/Discord/Internal/Rest/Webhook.hs:50]: I had a thought or two on that as well. the only place the auth is needed is in the restcallthread thing. as in the minimal.hs example you put up on discord a while ago, could we not expand support for webhook only messagin stuff by creating a separate one of those for noauth requests. means that most requests will just bounce, but gives people the option.

Comment (aquarial) [reply src/Discord/Internal/Rest/Webhook.hs]: I think using discord-haskell soley for an authenticated webhook http interface is an edge case. Creating a bot-application is pretty simple and unlocks everything else in the repo. If really desired minimal.hs is up as https://github.com/discord-haskell/discord-haskell/blob/master/examples/rest-without-gateway.hs

Comment (L0neGamer) [reply src/Discord/Internal/Rest/Webhook.hs]: I'm happy to remove the token cases if we think that they shouldn't be used.

Comment (aquarial) [reply src/Discord/Internal/Rest/Webhook.hs]: Yeah remove the webhook token from Get/Modify/Delete. (sending the token with ModifyWebhook [prevents changing the channelID](https://discord.com/developers/docs/resources/webhook#modify-webhook-with-token) the webhook runs in). 
Execute seems to require it though. 

Change that, and then you should have access to merge the PR

**Pull request merged**